### PR TITLE
Prevent rename of visible levels

### PR DESCRIPTION
**Pull request by davidsbailey**
# Description

Prevent rename of levels in a visible script. Also prevent rename of levels which are loaded when you go to a url like /projects/applab or /projects/dance . 

## Links

- [jira](https://codedotorg.atlassian.net/browse/LP-294)

## Testing story

Manually verified the following:

1. no change in behavior for level in a hidden script
Screen Shot 2019-11-25 at 6 44 20 PM

2. correct behavior + message for level in a visible script:
Screen Shot 2019-11-25 at 6 44 27 PM

3. correct behavior + message for a standalone project level:
Screen Shot 2019-11-25 at 6 44 35 PM

# Reviewer Checklist:

- [ ] Tests provide adequate coverage
- [ ] Code is well-commented
- [x] New features are translatable or updates will not break translations
- [x] Relevant documentation has been added or updated
- [x] User impact is well-understood and desirable
- [x] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

**Review by dmcavoy — APPROVED** [review]
Looks good Dave. There are gonna be merge conflicts between your changes and mine so I think I'm gonna pull this branch into my levelbuilder work.

**Comment by islemaster** [comment]
Is there a way to unit-test this? This seems like a bit of business logic we should cover. (Unless I've misunderstood the motivation for this change - it's not captured in the PR.)

File: dashboard/app/views/levels/_form.html.haml [hunk 1]
```diff
@@ -1,102 +0,0 @@
-- content_for(:head) do
-  %script{src: webpack_asset_path('js/levelbuilder.js')}
-
-%pre#validation-error.validation-error{style: 'background-color: yellow; display: none'}
```
Comment (dmcavoy) [inline dashboard/app/views/levels/_form.html.haml]: looks like you lost this line below. Not sure if intentional

Comment (davidsbailey) [reply dashboard/app/views/levels/_form.html.haml]: Sorry, yes, this was intentional. I did this in a separate commit with a descriptive name, but forgot to mention it in the PR description.

**Comment by davidsbailey** [comment]
> Is there a way to unit-test this? This seems like a bit of business logic we should cover. (Unless I've misunderstood the motivation for this change - it's not captured in the PR.)

Good call. I've added unit tests. The motivation, which is not really explained in the linked jira ticket, is to prevent levelbuilders from editing levels in cases where that could cause progress loss, which includes any time a script is visible. I added on the bit about standalone project levels because it seems like changing the name of those would break the corresponding standalone project page.

In explaining this I realized that I've failed to cover the case for pilot scripts, which are always marked hidden. I'll add that now. Thank you for prompting me for this!

**Review by islemaster — APPROVED** [review]
:+1: Thanks!

**Pull request merged**